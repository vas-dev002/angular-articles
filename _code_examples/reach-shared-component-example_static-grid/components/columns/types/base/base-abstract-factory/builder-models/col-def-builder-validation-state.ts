import { ColDefBuilderValidationItem } from "./col-def-builder-validation-item";

export class ColDefBuilderValidationState {
	colId = new ColDefBuilderValidationItem("Column id is not set", "Column id was already set");
	valueGetter = new ColDefBuilderValidationItem("Value getter is not set", "Value getter was already set");
	valueSetter = new ColDefBuilderValidationItem("Value setter is not set", "Value setter was already set");
	valueFormatter = new ColDefBuilderValidationItem("Value formatter is not set", "Value formatter was already set");
	sorting = new ColDefBuilderValidationItem("Sorting is not set", "Setter was already set");
	filtering = new ColDefBuilderValidationItem("Filtering is not set", "Filtering was already set");
	grouping = new ColDefBuilderValidationItem("Grouping is not set", "Grouping was already set");
	cellRendering = new ColDefBuilderValidationItem("Cell rendering component is not set", "Cell rendering component was already set");
	width = new ColDefBuilderValidationItem("Width is not set", "Width was already set");
	menuTabs = new ColDefBuilderValidationItem("Menu tabs are not set", "Menu tabs were already set");
	cellClass = new ColDefBuilderValidationItem("Cell class rules are not set", "Cell class rules were already set");
	headerName = new ColDefBuilderValidationItem("Header name is not set", "Header name was already set");
	visibility = new ColDefBuilderValidationItem("Visibility is not set", "Visibility was already set");
	tooltip = new ColDefBuilderValidationItem("Tooltip is not set", "Tooltip was already set");
}
