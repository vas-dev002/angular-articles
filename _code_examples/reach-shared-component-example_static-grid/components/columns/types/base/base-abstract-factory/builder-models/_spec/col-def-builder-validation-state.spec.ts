import { ColDefBuilderValidationState } from "../col-def-builder-validation-state";

declare const expect: jest.Expect;

describe("[Model]: ColDefBuilderValidationState", () => {
	let colDefBuilderValidationState: ColDefBuilderValidationState;

	beforeEach(() => {
		colDefBuilderValidationState = new ColDefBuilderValidationState();
	});

	describe("[ColDefBuilderValidationState instantiate]", () => {
		it("should be successfully created", () => {
			expect(colDefBuilderValidationState).toBeTruthy();
			expect(colDefBuilderValidationState).toMatchSnapshot();
		});
	});
});
