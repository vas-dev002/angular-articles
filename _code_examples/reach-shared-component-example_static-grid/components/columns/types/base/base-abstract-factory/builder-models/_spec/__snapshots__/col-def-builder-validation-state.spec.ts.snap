// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Model]: ColDefBuilderValidationState [ColDefBuilderValidationState instantiate] should be successfully created 1`] = `
ColDefBuilderValidationState {
  "cellClass": ColDefBuilderValidationItem {
    "duplicateErrorMessage": "Cell class rules were already set",
    "isSet": false,
    "requiredErrorMessage": "Cell class rules are not set",
  },
  "cellRendering": ColDefBuilderValidationItem {
    "duplicateErrorMessage": "Cell rendering component was already set",
    "isSet": false,
    "requiredErrorMessage": "Cell rendering component is not set",
  },
  "colId": ColDefBuilderValidationItem {
    "duplicateErrorMessage": "Column id was already set",
    "isSet": false,
    "requiredErrorMessage": "Column id is not set",
  },
  "filtering": ColDefBuilderValidationItem {
    "duplicateErrorMessage": "Filtering was already set",
    "isSet": false,
    "requiredErrorMessage": "Filtering is not set",
  },
  "grouping": ColDefBuilderValidationItem {
    "duplicateErrorMessage": "Grouping was already set",
    "isSet": false,
    "requiredErrorMessage": "Grouping is not set",
  },
  "headerName": ColDefBuilderValidationItem {
    "duplicateErrorMessage": "Header name was already set",
    "isSet": false,
    "requiredErrorMessage": "Header name is not set",
  },
  "menuTabs": ColDefBuilderValidationItem {
    "duplicateErrorMessage": "Menu tabs were already set",
    "isSet": false,
    "requiredErrorMessage": "Menu tabs are not set",
  },
  "sorting": ColDefBuilderValidationItem {
    "duplicateErrorMessage": "Setter was already set",
    "isSet": false,
    "requiredErrorMessage": "Sorting is not set",
  },
  "tooltip": ColDefBuilderValidationItem {
    "duplicateErrorMessage": "Tooltip was already set",
    "isSet": false,
    "requiredErrorMessage": "Tooltip is not set",
  },
  "valueFormatter": ColDefBuilderValidationItem {
    "duplicateErrorMessage": "Value formatter was already set",
    "isSet": false,
    "requiredErrorMessage": "Value formatter is not set",
  },
  "valueGetter": ColDefBuilderValidationItem {
    "duplicateErrorMessage": "Value getter was already set",
    "isSet": false,
    "requiredErrorMessage": "Value getter is not set",
  },
  "valueSetter": ColDefBuilderValidationItem {
    "duplicateErrorMessage": "Value setter was already set",
    "isSet": false,
    "requiredErrorMessage": "Value setter is not set",
  },
  "visibility": ColDefBuilderValidationItem {
    "duplicateErrorMessage": "Visibility was already set",
    "isSet": false,
    "requiredErrorMessage": "Visibility is not set",
  },
  "width": ColDefBuilderValidationItem {
    "duplicateErrorMessage": "Width was already set",
    "isSet": false,
    "requiredErrorMessage": "Width is not set",
  },
}
`;
